settings.js

```diff
@@ -109,6 +109,15 @@ function getWorkspaceSettingsByUUID(uuid) {
         path: `/org/gnome/shell/extensions/paperwm/workspaces/${uuid}/`});
 }
 
+function disconnectWorkspaceSettings(uuid) {
+    let list = workspaceList.get_strv('list');
+    let i = list.indexOf(uuid);
+    if (i > -1) {
+        list.splice(i, 1);
+        workspaceList.set_strv('list', list)
```

Comment:
Won't this result in losing all the workspace settings? Saved preferences are looked up through the list of uuids, so if they're no longer there we simply can't find them again.

Reply:
Yes, so it should either be combined with code that picks from existing "disconnected" settings when new workspaces are created. (PS: I see that the disconnectWorkspaceSettings is called for all workspaces - that's not intended ofc.)

Not sure if it's hard to enumerate all workspace settings (in the dconf database). Maybe it's simpler to set the index to -1 and without removing it from the list.

Reply:
Hmm, took the list of uuids pattern from gnome-terminal, I assumed it was basically required, though dconf-editor obviously manages to look them up, so it would be a bit weird if it was impossible when I think about it.